compiler/rustc_abi/src/layout.rs

```diff
@@ -426,10 +426,11 @@ where
                 }
             }
         }
-        _ => assert!(
-            start == Bound::Unbounded && end == Bound::Unbounded,
-            "nonscalar layout for layout_scalar_valid_range type: {st:#?}",
-        ),
+        _ => {
+            if start != Bound::Unbounded || end != Bound::Unbounded {
+                return None;
+            }
+        }
```

Comment:
Can be reverted, this is hit when `rustc_scalar_layout_start` is used on structs with a non-scalar field. We can just stop testing that. It's all unstable anyway.